lib/puppet/type/chocolateysource.rb

```diff
@@ -120,6 +111,10 @@ def insync?(is)
     if provider.respond_to?(:validate)
       provider.validate
     end
+
+    if (self[:ensure].to_sym == :present && (self[:location].nil? || self[:location] == ''))
+      raise ArgumentError, "A non-empty location must be specified when not disabling/removing a source."
```

Comment:
This error message is very confusing as it includes a double negative. Would prefer;
`A location must be specified when setting a source`